flux-kube/plugins/flux-kube.c

```diff
@@ -0,0 +1,284 @@
+                                 "flux-kube::free")) {
+        if (flux_jobtap_event_post_pack (p,
+                                         FLUX_JOBTAP_CURRENT_JOB,
+                                         "free",
+                                         NULL) < 0)
+             flux_log_error (flux_jobtap_get_flux (p),
+                             "flux-kube: failed to post free event");
+    }
+    return 0;
+}
+
+static int validate_cb (flux_plugin_t *p,
+                        const char *topic,
+                        flux_plugin_arg_t *args,
+                        void *arg)
+{
+    int fluxkube = 0;
+    json_t *jobspec = NULL;
+    char *js, *s;
+    uint32_t userid = (uint32_t) -1;
+
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i s:{s:{s:{s?i}}}}",
+                                "userid", &userid,
+                                "jobspec",
+                                 "attributes",
+                                  "system",
+                                   "flux-kube", &fluxkube) < 0) {
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "invalid system.flux-kube: %s",
+                                       flux_plugin_arg_strerror (args));
+    }
+
+    /*  Nothing to do if not fluxkube
+     */
+    if (!fluxkube)
+        return 0;
+
+    if (userid != getuid ())
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "Guest user cannot use alloc bypass");
+
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:o}",
+                                "jobspec", &jobspec) < 0) {
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "invalid jobspec: %s",
+                                       flux_plugin_arg_strerror (args));
+    }
+   /*  Store jobspec in job structure to avoid re-fetching from plugin args
+     *   in job.state.sched callback.
+     */
+    if (!(js = json_dumps (jobspec, 0))
+        || flux_jobtap_job_aux_set (p,
+                                    FLUX_JOBTAP_CURRENT_JOB,
+                                    "flux-kube::jobspec",
+                                    js,
+                                    free) < 0) {
+        free (js);
+        return flux_jobtap_reject_job (p,
+                                       args,
+                                       "failed to capture flux-kube "
+                                       "jobspec: %s",
+                                       strerror (errno));
+    }
+    /*  Store R string in job structure to avoid re-fetching from plugin args
+     *   in job.state.sched callback.
+     */
+    s = "{\"version\": 1, "
```

Comment:
Do you have a suggestion for the jansson call I could use?  The ones I've tried so far (`json_loads`, `json_string`) are just as hard to read (i.e., require all the escapes).

Reply:
Since this is a simple construction, `json_pack` would do it? For example, please see:

https://github.com/flux-framework/flux-sched/blob/master/resource/writers/match_writers.cpp#L206

Reply:
Perhaps someone with keener eyes can see what I've done wrong here for this example: `{"version": 1, "execution": {"R_lite": [{"rank": "0", "children": {"core": "0"}}], "starttime": 0.0, "expiration": 0.0, "nodelist": ["kubernetes"]} }`
```C
    R = json_pack ("{s:i s:{s:[{s:s s:{s:s}}] s:f s:f s:[s]}}", 
                  "version", 1, 
                  "execution", 
                         "R_lite",
                              "rank", "0",
                              "children",
                                     "core", "0"
                          "starttime", 0.0, 
                          "expiration", 0.0,
                          "nodelist", 
                                 "kubernetes");

    if (!(Rstring = json_dumps (R, 0))) {
```
`Rstring` is null.